pdpv0: panic in `processIndexingAndIPNICleanup` when creating IPNI removal advertisement

### Checklist

- [x] This is **not** a question or a support request. If you have any Curio related questions, please ask in the [discussion forum](https://github.com/filecoin-project/curio/discussions).
- [x] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/curio/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [x] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/curio/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [x] I **have** searched on the [issue tracker](https://github.com/filecoin-project/curio/issues) and the [discussion forum](https://github.com/filecoin-project/curio/discussions), and there is no existing related issue or discussion.
- [x] I am running the [`Latest release`](https://github.com/filecoin-project/curio/releases), or the most recent RC(release candidate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [x] I did not make any code changes to curio.

### Curio component

- [ ] PoRep Pipeline
- [ ] Snap Deals
- [ ] Market
- [x] IPNI provider
- [ ] Curio UI
- [x] Other

### Curio Version

```text
curio version 1.27.0+calibnet+git_6814fede_2025-11-20T11:57:49+04:00
```

### Describe the Bug

 `processIndexingAndIPNICleanup` panics when creating an IPNI removal advertisement because the `Entries` field is nil.

```
panic: interface conversion: datamodel.Link is nil, not cidlink.Link

github.com/ipni/go-libipni/ingest/schema.signaturePayload
    go-libipni@v0.7.1/ingest/schema/envelope.go:89
github.com/filecoin-project/curio/tasks/pdp.processIndexingAndIPNICleanup.func1
    tasks/pdp/watch_piece_delete.go:333
```

The `schema.Advertisement` is created without `Entries` at `watch_piece_delete.go:294-299`, but `Sign()` requires it.

## Workaround

Comment out `processIndexingAndIPNICleanup` call at `watch_piece_delete.go:47-50`.


### Logging Information

```text
panic: interface conversion: datamodel.Link is nil, not cidlink.Link

github.com/ipni/go-libipni/ingest/schema.signaturePayload
    go-libipni@v0.7.1/ingest/schema/envelope.go:89
github.com/filecoin-project/curio/tasks/pdp.processIndexingAndIPNICleanup.func1
    tasks/pdp/watch_piece_delete.go:333
```

### Repo Steps

https://filecoinproject.slack.com/archives/C095WFA0QK1/p1764087444250079

--- a/tasks/pdp/watch_piece_delete.go
+++ b/tasks/pdp/watch_piece_delete.go
@@ -296,6 +296,7 @@ func processIndexingAndIPNICleanup(ctx context.Context, db *harmonydb.DB, cfg *c
 				ContextID: contextID,
 				Metadata:  md,
 				IsRm:      true,
+				Entries:   schema.NoEntries,
 			}
 
 			{

